base/io.jl

```diff
@@ -794,38 +794,45 @@ function write(s::IO, A::AbstractArray)
         error("`write` is not supported on non-isbits arrays")
     end
     nb = 0
+    r = Ref{T}()
     for a in A
-        nb += write(s, a)
+        r[] = a
+        nb += @noinline unsafe_write(s, r, sizeof(r)) # r must be heap-allocated
     end
     return nb
 end
 
-function write(s::IO, a::Array)
-    if isbitstype(eltype(a))
-        return GC.@preserve a unsafe_write(s, pointer(a), sizeof(a))
-    else
+function write(s::IO, A::StridedArray)
+    if !isbitstype(eltype(A))
         error("`write` is not supported on non-isbits arrays")
     end
-end
-
-function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
-    if !isbitstype(T) || !isa(a, StridedArray)
-        return invoke(write, Tuple{IO, AbstractArray}, s, a)
+    elsz = sizeof(eltype(A))
```

Comment:
This is not the same as the zeroth stride of A, which is:
```suggestion
    elsz = elsize(typeof(A))
```
Though elsize seems to be implemented incorrectly for most AbstractArrays, so a better option is probably
```suggestion
    elsz = aligned_sizeof(T)
```

Reply:
Looks like `elsize` is better for `Base`, as `aligned_sizeof` seems wrong for `ReinterpretArray`.

Reply:
The result for `ReinterpretArray` is wrong also, for which the correct result is `packedsize(T)`

Reply:
Looks like `packedsize` is used only for bits `reinterpret`?
`ReinterpretArray` uses `sizeof` in `_getindex_ra` thus the padding won't be ignored.

Reply:
Oh weird. Alright then